worker/solution.go

```diff
@@ -0,0 +1,128 @@
+package worker
+
+import (
+	"fmt"
+	"os"
+	"errors"
+	"gorm.io/gorm"
+	"github.com/mini-maxit/worker/models"
+)
+
+
+// ErrInvalidFileCount is returned when the solution directory contains more than one solution file
+var ErrInvalidFileCount = errors.New("expected exactly one file in the solution directory")
+
+// ErrFileNotFound is returned when a file is not found
+var ErrFileNotFound = errors.New("file not found")
+
+// GetDataForSolutionRunner retrieves the data needed to run the solution
+func getDataForSolutionRunner(db *gorm.DB, task_id, user_id, user_solution_id, input_output_id int) (models.Task, error) {
+	var task models.Task
+	var dirPath string
+
+	// Fetch data with GORM
+	err := db.Raw(`
```

Comment:
We are using ORM, writing plain SQL is not what i would like to see. Use proper GORM constructoins to execute queries. Example [here](https://github.com/HermanPlay/WebApp/blob/master/backend/package/repository/user.go#L65-L76)